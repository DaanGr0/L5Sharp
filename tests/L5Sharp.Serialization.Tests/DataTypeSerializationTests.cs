using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using L5Sharp.Core;
using L5Sharp.Extensions;
using L5Sharp.Types;
using NUnit.Framework;

namespace L5Sharp.Serialization.Tests
{
    [TestFixture]
    public class DataTypeSerializationTests
    {
        [Test]
        public void Serialize_WhenCalled_ShouldNotBeNull()
        {
            var type = new DataType("Test", members: new[] { Member.Create("Test", new Dint()) });
            
            var element = type.Serialize();

            element.Should().NotBeNull();
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Serialize_WhenCalled_ShouldHaveApprovedOutput()
        {
            var type = new DataType("Test", members: new[] { Member.Create("Test", new Dint()) });
            
            var element = type.Serialize();

            Approvals.VerifyXml(element.ToString());
        }
        
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Serialize_WithMembers_ShouldHaveApprovedOutput()
        {
            var type = new DataType("Test", "My Test DataType", new List<IMember<IDataType>>
            {
                Member.Create("Member_Bool", new Bool()),
                Member.Create("Member_Sint", new Sint()),
                Member.Create("Member_Int", new Int()),
                Member.Create("Member_Dint", new Dint()),
                Member.Create("Member_Lint", new Lint()),
                Member.Create("Member_Real", new Real())
            });
            
            var element = type.Serialize();

            Approvals.VerifyXml(element.ToString());
        }
        
    }
}