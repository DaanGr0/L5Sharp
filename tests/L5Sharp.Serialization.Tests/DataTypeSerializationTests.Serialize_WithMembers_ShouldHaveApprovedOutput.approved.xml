<DataType Name="Test" Family="NoFamily" Class="User">
  <Description><![CDATA[My Test DataType]]></Description>
  <Members>
    <Member Name="Member_Bool" DataType="BOOL" Dimensions="0" Radix="Decimal" ExternalAccess="Read/Write" />
    <Member Name="Member_Sint" DataType="SINT" Dimensions="0" Radix="Decimal" ExternalAccess="Read/Write" />
    <Member Name="Member_Int" DataType="INT" Dimensions="0" Radix="Decimal" ExternalAccess="Read/Write" />
    <Member Name="Member_Dint" DataType="DINT" Dimensions="0" Radix="Decimal" ExternalAccess="Read/Write" />
    <Member Name="Member_Lint" DataType="LINT" Dimensions="0" Radix="Decimal" ExternalAccess="Read/Write" />
    <Member Name="Member_Real" DataType="REAL" Dimensions="0" Radix="Float" ExternalAccess="Read/Write" />
  </Members>
</DataType>